ai2_internal/citation_links/interface.py

```diff
@@ -0,0 +1,88 @@
+"""
+This file contains the classes required by Semantic Scholar's
+TIMO tooling.
+
+You must provide a wrapper around your model, as well
+as a definition of the objects it expects, and those it returns.
+"""
+
+from typing import List, Tuple
+
+from pydantic import BaseModel, BaseSettings
+
+from mmda.predictors.xgb_predictors.citation_link_predictor import CitationLinkPredictor
+from mmda.types import api
+from mmda.types.document import Document
+
+# these should represent the extracted citation mentions and bibliography entries for a paper
+class Instance(BaseModel):
+    """
+    Describes one Instance over which the model performs inference.
+    """
+    symbols: str
+    mentions: List[api.SpanGroup]
+    bibs: List[api.SpanGroup]
+
+
+class Prediction(BaseModel):
+    """
+    Describes the outcome of inference for one Instance
+    """
+    linked_mentions: List[Tuple[api.SpanGroup, str]]
```

Comment:
suggestion: we just return List[(str,str)] here too? we can make that structure back on the client.

Reply:
works for me!